TYPO3.Neos/Classes/TYPO3/Neos/Controller/Frontend/NodeController.php

```diff
@@ -157,6 +157,8 @@ protected function handleShortcutNode(NodeInterface $node)
             throw new NodeNotFoundException(sprintf('The shortcut node target of node "%s" could not be resolved', $node->getPath()), 1430218730);
         } elseif (is_string($resolvedNode)) {
             $this->redirectToUri($resolvedNode);
+        } elseif ($resolvedNode instanceof NodeInterface && $resolvedNode === $node) {
```

Comment:
$node is already an instance of `NodeInterface` so the `instanceof` check is not needed here, is it?
```php
} elseif ($resolvedNode === $node) {
```
should do

Reply:
But the check is against `$resolvedNode`…

Reply:
But if `$node` is an instance of `NodeInterface` the second part of the condition already implies the type.. Anyways, not a biggy